index.html

```diff
@@ -68,6 +68,22 @@ <h1 class="text-l"><a href="https://amche.in" class="text-white hover:underline"
     </main>
 
     <script>
+
+//FIXME: move this function to a utils
+function getQueryParameters() {
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const params = {};
+
+    // Iterate through all entries in the URLSearchParams object
+    for (const [key, value] of urlParams.entries()) {
+        params[key] = value;
+    }
+
+    return params; // Return the object containing all query parameters
+}
```

Comment:
nit: can `Object.fromEntries` come handy here? `Object.fromEntries(new URLSearchParams(window.location.search).entries());`

Reply:
Sorry been traveling and missed getting to this - good catch and makes sense - can make this change in a separate PR / alongwith a refactor.